packages/rest/test/acceptance/validation/validation.acceptance.ts

```diff
@@ -65,7 +65,10 @@ describe('Validation at REST level', () => {
     it('rejects requests with no (empty) body', async () => {
       // NOTE(bajtos) An empty body cannot be parsed as a JSON,
       // therefore this test request does not even reach the validation logic.
-      await client.post('/products').expect(400);
+      await client
+        .post('/products')
+        .type('json')
+        .expect(422);
```

Comment:
With this change in place, the comment at the top of the test is inconsistent with what is the test asserting. Please take a closer look, I would like us to understand how are empty payloads handled with the new request handler in place.

Reply:
PTAL ☝️

Reply:
Still not addressed ☝️

Reply:
It turns out that express json body parser creates `{}` for empty body. I added some logic in json parser to treat empty body as `undefined`.